python/lsst/ip/isr/isrMockLSST.py

```diff
@@ -135,63 +130,68 @@ def makeImage(self):
             if self.config.isTrimmed:
                 bbox = amp.getBBox()
             else:
-                bbox = amp.getRawDataBBox().shiftedBy(amp.getRawXYOffset())
+                bbox = amp.getRawDataBBox()
 
             ampData = exposure.image[bbox]
 
             # Sky effects in e-
             if self.config.doAddSky:
-                self.amplifierAddNoise(ampData, self.config.skyLevel, np.sqrt(self.config.skyLevel))
+                self.amplifierAddNoise(ampData, self.config.skyLevel * self.config.gain,
+                                       np.sqrt(self.config.skyLevel * self.config.gain))
 
             if self.config.doAddSource:
                 for sourceAmp, sourceFlux, sourceX, sourceY in zip(self.config.sourceAmp,
                                                                    self.config.sourceFlux,
                                                                    self.config.sourceX,
                                                                    self.config.sourceY):
                     if idx == sourceAmp:
-                        self.amplifierAddSource(ampData, sourceFlux, sourceX, sourceY)
+                        self.amplifierAddSource(ampData, sourceFlux * self.config.gain, sourceX, sourceY)
 
             # Other effects in e-
             if self.config.doAddFringe:
-                self.amplifierAddFringe(amp, ampData, np.array(self.config.fringeScale),
+                self.amplifierAddFringe(amp, ampData, np.array(self.config.fringeScale) * self.config.gain,
                                         x0=np.array(self.config.fringeX0),
                                         y0=np.array(self.config.fringeY0))
 
             if self.config.doAddFlat:
                 if ampData.getArray().sum() == 0.0:
+                    # Add noise
```

Comment:
What is happening here?  The call below isn't adding noise, but is adding 1 electron and no noise.  @czwa?

Reply:
I think this was my hack for what is now better controlled by the `calibMode`.  If you're making a combined flat, you don't want any of the previous signals included, so mock that reaches this point that is universally 0.0 (which this isn't the best check for, but it's unlikely that a populated mock will sum to zero) and has `doAddFlat=True` is probably making the combined flat.  In this case, the mock should be filled with 1.0 so that the flat signal can be included.
This should probably instead be a check for `if self.config.calibMode:`.